JOSESwift/Sources/ECKeys.swift

```diff
@@ -187,6 +197,14 @@ public struct ECPublicKey: JWK {
 
         return try T.representing(ecPublicKeyComponents: (self.crv.rawValue, x, y))
     }
+
+    @available(iOS 11.0, *)
+    public func withKeyIdFromThumbprint(algorithm: ThumbprintAlgorithm = .SHA256) throws -> Self {
```

Comment:
We need to add a documentation comment to the `withKeyIdFromThumbprint` functions. To avoid having to add the comment to all versions of the function you could make this an internal helper function and then call the helper functions once in a public extension to JWK.

Reply:
Pushed some changes now :)